### DISCUSSION TITLE
Are there test cases for the software

### DISCUSSION THREAD

**Discussion by 123xyl** [openmopac/mopac#123] [Q&A]
Are there test cases for the software

**Comment by godotalgorithm** [comment]
Yes, there is a `tests` directory of the MOPAC repository that contains a set of tests, with 66% code coverage right now. These tests can be run using CTest, which is a feature of CMake. Once MOPAC has been built, you can run these tests using the command `ctest` from the MOPAC build directory.

**Reply by 123xyl** [reply]
When I install, I directly create a new build directory in the mopac source directory, and then perform cmake, make, make, and install in the build directory. Do I run ctest directly in the build directory

**Reply by 123xyl** [reply]
When I run ctest in build, all are failed

**Reply by godotalgorithm** [reply]
You run ctest in the build directory containing the successfully compiled executable. It also has a verbose mode, `ctest --verbose`, that provides a more detailed error message if a test fails. The testing scripts require python and numpy to be installed as well, which is the most likely reason for the tests failing.

**Reply by 123xyl** [reply]
Are the two regenerate_tests.sh in the source file examples and can be run directly

**Reply by godotalgorithm** [reply]
`regenerate_tests.sh` is a shell script for recreating the reference data used for comparisons in the regression tests. You should not run `regenerate_tests.sh`. All of the `*.mop` files in the test directory are examples of MOPAC input files that are used by CTest in the testing process.

**Reply by 123xyl** [reply]
thank you very much
